rfcs/text/0000-extend-file-pe.md

```diff
@@ -0,0 +1,1076 @@
+# 0000: Extend the PE subfield of the File fieldset
+
+- Stage: **1 (proposal)**
+- Date: **TBD**
+
+The Portable Executable (PE) sub-field, of the `file` top-level fieldset, can be updated to include more file attributes to aid in file analysis. This additional document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+**Stage 0**
+
+This RFC is to create 25 additional sub-fields within the `file.pe` fieldset.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| file.pe.authentihash | keyword | Authentihash of the PE file. |
+| file.pe.compile_timestamp | date | Compile timestamp of the PE file. |
+| file.pe.compiler_product_versions | keyword | Version of the compiler. |
+| file.pe.creation_date | date | Extracted when possible from the file's metadata. Indicates when it was built or compiled. It can also be faked by malware creators. |
+| file.pe.entry_point | long | Entry point of the PE file. |
+| file.pe.exports | keyword | List of symbols exported by PE |
+| file.pe.debug | flattened | Debug information, if present |
+| file.pe.import_list | flattened | List of all imported functions |
+| file.pe.sections | flattened | Data about sections of compiled binary PE |
+| file.pe.resource_details | flattened | If the PE contains resources, some info about them |
+| file.pe.resource_languages | flattened | Digest of languages found in resources. Key is language (as string) and value is how many resources there are having that language (as integer) |
+| file.pe.resource_types | flattened | Digest of resource types. Key is resource type (as string) and value is how many resources there are of that specific type (as integer) |
+| file.pe.packers | flattened | Identifies packers used on Windows PE files by several tools and AVs. Keys are tool names and values are identified packers, both strings. See `file.pe.packers` for merged list of packers from all tools. |
+| file.pe.machine_type | keyword | Machine type of the PE file. |
+| file.pe.main_icon.hash.dhash | keyword | Difference Hash for a given PE file. |
+| file.pe.main_icon.hash.md5 | keyword | MD5 hash of raw icon data |
+| file.pe.overlay.chi2 | float | Chi2 information of the PE file. |
+| file.pe.overlay.entropy | float | Entropy information of the PE file. |
```

Comment:
It is calculated. In this case calculated by VT, but many PE analyzers also do this. The chi2 calculation is also useful, as it provides a sense of frequency distribution of data.

Reply:
@webmat this is generally the [Shannon Entropy](https://en.wikipedia.org/wiki/Entropy_(information_theory)) of a blob of data. I'm thinking it might be worth linking to an article on the algorithm or even just this wiki article, but it's a well-known algorithm, so a link is probably enough.